core/src/main/java/io/grpc/ChannelImpl.java

```diff
@@ -119,6 +125,10 @@ public synchronized ChannelImpl shutdown() {
       return this;
     }
     shutdown = true;
+    if (deadlineCancellationExecutor != null) {
+      deadlineCancellationExecutor =
```

Comment:
to check my work a little more: if `shutdown`, we'll never obtain an active transport in `start`, and therefore will enter `closeCallPrematurely`, rather than attempting to begin a new cancellation task.